src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -4692,6 +4692,26 @@ public override IOperation VisitInvocation(IInvocationOperation operation, int?
             return (visitedInstance, visitedArguments);
         }
 
+        internal override IOperation VisitNoPiaObjectCreation(INoPiaObjectCreationOperation operation, int? argument)
+        {
+            // Initializer is removed from the tree and turned into a series of statements that assign to the created instance
+            IOperation initializedInstance = new NoPiaObjectCreationOperation(initializer: null, semanticModel: null, operation.Syntax, operation.Type,
+                                                                              operation.ConstantValue, IsImplicit(operation));
+
+            if (operation.Initializer != null)
```

Comment:
I believe @333fred is [pulling out](https://github.com/dotnet/roslyn/pull/27460/files#diff-2c0a650e29227920ce474a68cd354d0eL4704) similar blocks of code from individual Visit methods into `HandleObjectOrCollectionInitializer`. You may want to wait for his change to go in and simplify this code?

Reply:
Yes, Aleksey said he'd move correct that once my PR is merged, which it is now. https://github.com/dotnet/roslyn/pull/27476#issuecomment-394776622